### ISSUE TITLE
[BUG][csharp] [csharp-netcore] producing invalid Dictionary

### ISSUE THREAD

**Issue by InTheCloudDan** [OpenAPITools/openapi-generator#9727]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### openapi-generator version

`5.1.1`
and
`5.2.0-20210607.065944-108`

##### Steps to reproduce
```
      "example": {
        "type": "object",
        "additionalProperties": true
      }
```
or `additionalProperties: {}`

Create a spec with the model above and use `csharp` or `csharp-netcore` to generate a client. It errors with `error CS0305: Using the generic type 'Dictionary<TKey, TValue>' requires 2 type arguments`

The property in code shows as `Dictionary` without any additional type information

**Comment by prajon84** [comment]
Thanks @InTheCloudDan . I also recently got stumbled in the similar issue.

**Comment by InTheCloudDan** [comment]
The one I'm running into seems like it originating from here: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/csharp/modelGeneric.mustache#L113

But with the same root cause, I found one that was also a `List` not just `Dictionary`

**Comment by InTheCloudDan** [comment]
@prajon84 no, we have not made any progress on this issue yet.

**Comment by prajon84** [comment]
> @prajon84 no, we have not made any progress on this issue yet.

Oh Ok. Thank you for the update on your side.

**Comment by prajon84** [comment]
@wing328 @Blackclaws @shibayan @lucamazzanti @devhl-labs  any idea to resolve this ?

**Comment by Blackclaws** [comment]
Yeah ok, that is a weird issue. Additional properties in general are a bit of a mixed bag. We're just throwing the bunch of them into a dictionary that should be string, object. I have no clue why it generates a List object type here or a dictionary without any types whatsoever. I'll take a look after the holiday season is over.

**Comment by almsyx** [comment]
Any progress on this? We're experiencing exactly the same...

**Comment by dietermijle** [comment]
Any progress on this one? We're experiencing the same issue...

**Comment by prajon84** [comment]
Hello @Blackclaws, any idea how to resolve this?
